### PR TITLE
ba-137 Retrieve categories from Firebase

### PR DESCRIPTION
**Pull request by lyio**
- Remove firebase collection for categories again
- Move clearing and setting of items to DialogBehavior
- Hide 'Best before' in `socobo-inventory-show` when no date is set
- Replace category dropdown with a quick select option for the expiry date

### CODE REVIEW

File: app/elements/socobo-inventory/socobo-inventory-dialog-behavior.html [hunk 1]
```diff
@@ -1,3 +1,4 @@
+<script type="application/javascript" src="../../bower_components/lodash/lodash.min.js"></script>
```
Comment (flashback2k14) [inline app/elements/socobo-inventory/socobo-inventory-dialog-behavior.html]: please remove the import

File: app/elements/socobo-inventory/socobo-inventory-validate-behavior.html [hunk 2]
```diff
@@ -37,14 +37,21 @@
...
        *
        * @private
        */
-      categories: {
+      dateOffsets: {
         type: Array,
         value: function() {
-          return ["Backen", "Gewürze", "Gemüse", "Obst"];
+          return [
+            this.Offset("none", 0, "day"),
```
Comment (flashback2k14) [inline app/elements/socobo-inventory/socobo-inventory-validate-behavior.html:48]: i think "none" should reset the date to today!?

**Comment by flashback2k14** [comment]
- close show dialog shows following error:
  bug 1
- best before has a "little" offset on the show dialog:
  bug 2

File: app/elements/socobo-inventory/socobo-inventory-dialog.html [hunk 3]
```diff
@@ -128,25 +141,27 @@
...
-                <template is="dom-repeat" items="[[categories]]" as="category">
-                  <paper-item c="[[category]]">[[category.value]]</paper-item>
+                <template is="dom-repeat" items="[[dateOffsets]]" as="offset">
+                  <paper-item c="[[offset]]">[[offset.value]]</paper-item>
                 </template>
               </paper-menu>
             </paper-dropdown-menu>
+            <paper-input
+              type="date"
+              class="style-date-picker"
```
Comment (flashback2k14) [inline app/elements/socobo-inventory/socobo-inventory-dialog.html]: double class attribute

File: app/elements/socobo-inventory/socobo-inventory-dialog.html [hunk 4]
```diff
@@ -63,15 +63,28 @@
...
+        margin-left: 16px;
+      }
+
       .style-dropdown-category {
         margin-top: 2px;
       }
+
+      .dropdown-content {
+        position: relative;
+        top: 0px;
```
Comment (flashback2k14) [inline app/elements/socobo-inventory/socobo-inventory-dialog.html]: px not needed with 0

File: app/elements/socobo-inventory/socobo-inventory-validate-behavior.html (hunk 2 above)
Comment (lyio) [inline app/elements/socobo-inventory/socobo-inventory-validate-behavior.html:48]: But that would make no sense. `None` should set the expiry date to _none_. That is, clear the input.

**Comment by patrick-hebner** [comment]
The expire date quick selection does not wrap correctly but I think this was noticed already.
Beside this and the points mentioned by @flashback2k14 (expect of the 'none' issue) that this PR works fine for me.

**Comment by lyio** [comment]
The points noted by @flashback2k14 should be fixed now. Also, the dropdown text wrapping was fixed by @flashback2k14. 

Thx for that :+1:

**Comment by flashback2k14** [comment]
Master of CSS ;-P

**Comment by lyio** [comment]
Yes!

**Comment by patrick-hebner** [comment]
lgtm

**Comment by flashback2k14** [comment]
lgtm

**Pull request merged**